indexbuilder.go

```diff
@@ -245,6 +249,18 @@ type DocumentSection struct {
 	Start, End uint32
 }
 
+type Symbol struct {
+	Name       string
+	Kind       string
```

Comment:
anyway to make this an enum and more mem efficient?

Reply:
Yep, a `[]string` can be added to the index file for `Kind`s along with the offsets so we can use a single `uint8` to indicate which `Kind.

Reply:
Can't the kind just be an enum? IE ctags has a static list of kind strings.